lib/OnyxCache.js

```diff
@@ -191,6 +191,16 @@ class OnyxCache {
     setRecentKeysLimit(limit) {
         this.maxRecentKeysSize = limit;
     }
+
+    /**
+     * @param {String} key
+     * @param {*} value
+     * @returns {Boolean}
+     */
+    hasValueChanged(key, value) {
+        this.addToAccessedKeys(key);
+        return !_.isEqual(this.storageMap[key], value);
```

Comment:
Did you by any chance explore other deep equal solutions? If we're doing this for performance reasons, then maybe we should try to find the fastest one?

Reply:
I explored this one [`dequal`](https://github.com/lukeed/dequal) and it does seem faster. But decided to use `_.isEqual()` since we already have the underscore dependency.

Reply:
So I went and benchmarked a few of these today and tried to compare what I think are our biggest data types here's what I came back with:

**Comparing two sets of very large `personalDetails` with one user displayName changed**

```
_.isEqual() x 1,248 ops/sec ±0.43% (96 runs sampled)
dequal() x 277 ops/sec ±2.30% (86 runs sampled)
deepEqual() x 1,231 ops/sec ±0.65% (93 runs sampled)
fastDeepEqual() x 1,130 ops/sec ±0.84% (93 runs sampled)
microDiff() x 214 ops/sec ±1.58% (87 runs sampled)
Fastest is _.isEqual()
```

**Comparing two `report` objects one with a different lastVisitedTimestamp**

```
_.isEqual() x 1,857,706 ops/sec ±0.17% (92 runs sampled)
dequal() x 2,556,428 ops/sec ±0.35% (93 runs sampled)
deepEqual() x 2,457,190 ops/sec ±0.33% (96 runs sampled)
fastDeepEqual() x 1,768,857 ops/sec ±0.09% (99 runs sampled)
microDiff() x 795,347 ops/sec ±0.19% (96 runs sampled)
Fastest is dequal()
```

**Comparing two `reportActions` objects each with 250 comments and one with an additional comment**
```
_.isEqual() x 258,482 ops/sec ±0.39% (95 runs sampled)
dequal() x 3,394 ops/sec ±0.47% (97 runs sampled)
deepEqual() x 260,473 ops/sec ±0.98% (94 runs sampled)
fastDeepEqual() x 268,251 ops/sec ±0.33% (96 runs sampled)
microDiff() x 1,828 ops/sec ±0.41% (99 runs sampled)
Fastest is fastDeepEqual()
```

So going off of these numbers it looks to me like:

- `dequal` is trailing hard for _very_ large objects, but best at small stuff
- `_.isEqual` seems pretty comparable to other libs at the large stuff

Based on that I think it's not really worth using `dequal` or any of the others. I'd be more concerned about performance with these really large objects and underscore isn't too far behind the competition.

Reply:
deepEqual looks like the best option here, no?
Given this is super low level code that will get called everywhere, I think is worth to add a new dependency for the sake of performance

Reply:
Cool I'm good with that. I wasn't sure if there was enough of a meaningful difference between that and `_.isEquals()`